tests/legacy-cli/e2e/tests/generate/library/library-consumption-linker.ts

```diff
@@ -0,0 +1,100 @@
+import { writeFile } from '../../../utils/fs';
+import { ng } from '../../../utils/process';
+import { updateJsonFile } from '../../../utils/project';
+
+export default async function () {
+  await ng('generate', 'library', 'my-lib');
+
+  // Enable partial compilation mode (linker) for the library
+  await updateJsonFile('projects/my-lib/tsconfig.lib.json', config => {
+    const { angularCompilerOptions = {} } = config;
+    angularCompilerOptions.compilationMode = 'partial';
```

Comment:
The partial should only be set in the production configuration.

This is because, during development Ivy will run in full mode, to avoid the linker step for faster incremental builds. (At least that was the idea before).

```suggestion
```

Reply:
I initially did this for the test to validate it would work with both types of configuration option sets (the ivy development scenario is already tested in the existing consumption test).  While we most likely won't generate an initial library this way, it should still work if a user chooses to do so.  I'm open to changing it though.

Reply:
In that case there isn’t any difference between a library built in dev or prod mode.

Only the consumption would be different.

I am fine either way.

Reply:
Updated to align with the recommended future library use cases.